packages/@lwc/engine/src/framework/decorators/wire.ts

```diff
@@ -55,3 +58,43 @@ export default function wire(_adapter: any, _config: any): DecoratorFunction {
         throw new TypeError();
     }
 }
+
+function createWiredPropertyDescriptor(
+    Ctor: ComponentConstructor,
+    key: string,
+    enumerable: boolean
+): PropertyDescriptor {
+    return {
+        get(this: ComponentInterface): any {
+            const vm = getComponentVM(this);
+            if (process.env.NODE_ENV !== 'production') {
+                assert.isTrue(vm && 'cmpRoot' in vm, `${vm} is not a vm.`);
+            }
+            valueObserved(this, key);
+            return vm.cmpFields[key];
+        },
+        set(this: ComponentInterface, newValue: any) {
+            const vm = getComponentVM(this);
+            if (process.env.NODE_ENV !== 'production') {
+                assert.isTrue(vm && 'cmpRoot' in vm, `${vm} is not a vm.`);
+                assert.invariant(
+                    !isRendering,
+                    `${vmBeingRendered}.render() method has side effects on the wired field ${vm}.${String(
+                        key
+                    )}`
+                );
+            }
+            // making the wired value as readonly
+            newValue = reactiveMembrane.getReadOnlyProxy(newValue);
```

Comment:
@davidturissini this is the meat of it. Track uses `reactiveMembrane.getProxy` while this is now using `reactiveMembrane.getReadOnlyProxy`